lib/brook/serializer.ex

```diff
@@ -0,0 +1,49 @@
+defprotocol Brook.Event.Kafka.Serializer do
+  @type type :: atom()
+  @type reason :: term()
+  @fallback_to_any true
+
+  @spec serialize(term()) :: {:ok, type(), term()} | {:ok, term()} | {:error, reason()}
+  def serialize(data)
+end
+
+defprotocol Brook.Event.Kafka.Deserializer do
+  @type reason :: term()
+  @fallback_to_any true
+
+  @spec deserialize(module(), term()) :: {:ok, term()} | {:error, reason()}
+  def deserialize(struct, data)
+end
+
+defimpl Brook.Event.Kafka.Serializer, for: Any do
+  def serialize(%custom_struct{} = data) do
```

Comment:
Didn’t realize you could pattern match on the struct without doing the `%{__struct__: type} = data`
Nice to know!